components/camera/server.go

```diff
@@ -91,12 +91,20 @@ func (s *subtypeServer) GetImage(
 			release()
 		}
 	}()
-
 	actualMIME, _ := utils.CheckLazyMIMEType(req.MimeType)
+	// if returned img is a LazyEncodedImage, it may have changed the MIMEType
+	if lazy, ok := img.(*rimage.LazyEncodedImage); ok {
+		lazyMIME := lazy.MIMEType()
+		if lazyMIME != actualMIME {
+			s.logger.Debugw("got different MIME type than what was asked for", "requested", actualMIME, "received", lazyMIME)
+		}
+		actualMIME = lazyMIME
```

Comment:
Why are we encoding based on what the response gave us? Shouldn't we encode based on what the request wants?

Reply:
Doesn't that add another decode-re-encode step? And does the user 
Like, let's say the user requests JPEG, because they want the transfer will be fast, but the server doesn't have JPEG, it only has PNG. So the server recieves a PNG, it has to be decoded, and then re-encoded as JPEG, and sent over to the client.

Here is the flow:
1. Server calls ReadImage on a camera.
2. camera is a client, so it calls the GetImage Request with a JPEG+lazy hint.
3. downstream server doesn't have JPEG, and returns a PNG instead.
4. PNG gets put in a LazyEncodedImage, since Lazy was requested.

Now we have two options:
Option1: Since the request was for JPEG, LazyEncodedImage has to be decoded anyway, and re-encoded as a JPEG. This ruins any speed up the user was expecting, but at least they get JPEG bytes back.
Option2: How the image gets encoded doesn't matter, and anyway, what the user requests might not always be what the server gives back. Since the downstream server gave a PNG, change the response MIME to PNG and return the PNG bytes.

Reply:
We should assume lazy is going away here. But I think it's a good question: if the user requests a particular mime type, and the camera (which may be an arbitrary remote/module) doesn't have it, should we decode/re-encode to the requested type. The fact that we return mime type in the response makes me think we should not, since that indicates we do not guarantee returning the requested mime type. This is also a Product question, so Bijan, I think you should loop in Fahmina.

Reply:
As long as mime is here, yes, we should always encode to what the user asked for, if we can.